### PR TITLE
@Added restArguments, isMap, isWeakMap, isSet, isWeakSet, toPath interface definitions to types/underscore

### PR DESCRIPTION
**Pull request by xiaoxizi2303**
@Added restArguments, isMap, isWeakMap, isSet, isWeakSet, toPath interface definitions to types/underscore

### CODE REVIEW

**Review by peterblazejewicz — COMMENTED** [review]
Hey,
this is extremely (still) popular library behind the types. It's kind of expected to provide at least basic tests as per README:
https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#my-package-teststs
(if there are not tests, maintainers have to give final approve to PR)
thanks!

File: types/underscore/index.d.ts
Comment (jgonggrijp) [inline types/underscore/index.d.ts:4916]: Could this type not be more precise? `_.restArguments` modifies the parameter list of a function, and I think TS might be able to express this.

Also, why did you add this to the `UnderscoreStatic` interface but not to the other two?

File: types/underscore/index.d.ts
Comment (jgonggrijp) [inline types/underscore/index.d.ts:5304]: Why did you add this to the `Underscore` and `_Chain` interfaces but not to `UnderscoreStatic`?

**Review by jgonggrijp — CHANGES_REQUESTED** [review]
Thanks for contributing. The changes mostly look correct to me, but I cannot approve without tests.

File: types/underscore/index.d.ts
Comment (xiaoxizi2303) [reply types/underscore/index.d.ts]: I will report an error when using it on my end, and it has been corrected so far

File: types/underscore/index.d.ts
Comment (xiaoxizi2303) [reply types/underscore/index.d.ts]: I will report an error when using it on my end, and it has been corrected so far

File: types/underscore/index.d.ts
Comment (xiaoxizi2303) [reply types/underscore/index.d.ts]: Modified

Comment (xiaoxizi2303) [reply types/underscore/index.d.ts]: Modified

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/underscore/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4ea98a48813ea79f11bd4b8e24556f96dc6ce273
</p>

File: types/underscore/index.d.ts
Comment (jgonggrijp) [inline types/underscore/index.d.ts:1021]: I think `path` by default is `number | string | (number | string)[]`. The return value must be `(number | string)[]`.

File: types/underscore/index.d.ts
Comment (jgonggrijp) [inline types/underscore/index.d.ts:3510]: No, `func` is the function to wrap.

**Review by jgonggrijp — CHANGES_REQUESTED** [review]
Please add tests.

Also, you flipped the situation with `toPath`. On your first submission, it was included in the `Underscore` and the `_Chain` interfaces, but not in `UnderscoreStatic`. Now, it is *only* in `UnderscoreStatic` but not in the other two. It needs to be in all three, just like `restArguments`.

File: types/underscore/index.d.ts
Comment (xiaoxizi2303) [reply types/underscore/index.d.ts:3510]: Are you saying that the func function is not parsed correctly in the comment?

File: types/underscore/index.d.ts
Comment (xiaoxizi2303) [reply types/underscore/index.d.ts:1021]: Are these the return values of the toPath (path:any) function? number | string | (number | string)[](number | string)[]

Comment (jgonggrijp) [reply types/underscore/index.d.ts:1021]: I mean the type should be like this:

```suggestion
        toPath(path: number | string | (number | string)[]): (number | string)[];
```

File: types/underscore/index.d.ts
Comment (jgonggrijp) [reply types/underscore/index.d.ts:3510]: Maybe? I am not sure I understand your question correctly.

Could you rephrase the question, and maybe write it in Mandarin Chinese as well?

Translation below.

我用谷歌翻译翻译了这个：

或许？ 我不确定我是否正确理解你的问题。

您能否重新表述一下这个问题，或者也用普通话写一下？

Comment (xiaoxizi2303) [reply types/underscore/index.d.ts:3510]: 您上面指出的问题是说我注释里面的内容写的不对吗

Comment (jgonggrijp) [reply types/underscore/index.d.ts:3510]: Yes, the problem is only in the comment. The type annotation is fine.

是的，问题只出在评论中。 类型注释很好。

Comment (xiaoxizi2303) [reply types/underscore/index.d.ts:3510]: Could you tell me what the correct content is here? Thank you, sir.

Comment (jgonggrijp) [reply types/underscore/index.d.ts:3510]:
```suggestion
         * @param func The function to wrap.
```

Comment (xiaoxizi2303) [reply types/underscore/index.d.ts:3510]: I see.

Comment (xiaoxizi2303) [reply types/underscore/index.d.ts:3510]: I see. Thank you.

File: types/underscore/index.d.ts
Comment (jgonggrijp) [inline types/underscore/index.d.ts:4937]: The first argument is implicit in the `Underscore` and `_Chain` interfaces.

第一个参数隐含在“Underscore”和“_Chain”接口中。

```suggestion
        /**
         * Wrapped type `(...args: any[]) => any`.
         * @see _.restArguments
         */
        restArguments(startIndex:number): (...args: any[]) => any;

        /**
         * Wrapped type `number | string | Array<number | string>`.
         * @see _.toPath
         */
        toPath(): Array<number | string>;
```

File: types/underscore/index.d.ts
Comment (jgonggrijp) [inline types/underscore/index.d.ts:6230]: The first argument is implicit in the Underscore and _Chain interfaces.

第一个参数隐含在“Underscore”和“_Chain”接口中。

```suggestion
        /**
         * Wrapped type `(...args: any[]) => any`.
         * @see _.restArguments
         */
        restArguments(startIndex:number): _Chain<T>;

        /**
         * Wrapped type `number | string | Array<number | string>`.
         * @see _.toPath
         */
        toPath(): _Chain<T>;
```

**Review by jgonggrijp — CHANGES_REQUESTED** [review]
Please add unittests.

请添加单元测试。

**Review by gabritto — CHANGES_REQUESTED** [review]
See review comments https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69550/files#r1610574218 and please update the existing tests to use the new methods.

**Comment by xiaoxizi2303** [comment]
> 请参阅 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69550/files#r1610574218 的审阅注释，请更新现有测试以使用新方法。
Okay, I see

**Pull request closed without merging**
